meshroom/ui/app.py

```diff
@@ -105,12 +117,24 @@ def __init__(self, args):
         # request any potential computation to stop on exit
         self.aboutToQuit.connect(r.stopChildThreads)
 
-        parser = argparse.ArgumentParser(prog=args[0], description='Launch Meshroom UI.')
-        parser.add_argument('--project', metavar='MESHROOM_FILE', type=str, required=False,
-                            help='Meshroom project file (e.g. myProject.mg).')
-        args = parser.parse_args(args[1:])
+        if args.pipeline:
+            # the pipeline from the command line has the priority
+            r.setDefaultPipeline(args.pipeline)
+        else:
+            # consider the environment variable
+            defaultPipeline = os.environ.get("MESHROOM_DEFAULT_PIPELINE", "")
+            if defaultPipeline:
+                r.setDefaultPipeline(args.pipeline)
+
         if args.project:
-            r.loadUrl(QUrl.fromLocalFile(args.project))
+            r.load(args.project)
+
+        if args.input:
```

Comment:
shouldn't `input` and `project` be exclusive?

At least if `project` and `input` both contain a .mg. In this case, I think it would be better to throw an error to let the user know that there is something wrong. (unless it is able to load both projects?)

And if we give `project` and `input` with folders, does it add the images to the current project? Something in the same spirit of meshroom_photogrammetry https://github.com/alicevision/meshroom/pull/632/files#diff-4f70151d95f3b3a1c7ea377cbaef3f10R114
```python
# initialize photogrammetry pipeline
if args.pipeline:
    # custom pipeline
    graph = meshroom.core.graph.loadGraph(args.pipeline)
    cameraInit = getOnlyNodeOfType(graph, 'CameraInit')
    # reset graph inputs
    cameraInit.viewpoints.resetValue()
    cameraInit.intrinsics.resetValue()
    # add views and intrinsics (if any) read from args.input
    cameraInit.viewpoints.extend(views)
    cameraInit.intrinsics.extend(intrinsics)

    if not graph.canComputeLeaves:
        raise RuntimeError("Graph cannot be computed. Check for compatibility issues.")

    if args.output:
        publish = getOnlyNodeOfType(graph, 'Publish')
        publish.output.value = args.output
else:
    # default pipeline
    graph = multiview.photogrammetry(inputViewpoints=views, inputIntrinsics=intrinsics, output=args.output)
    cameraInit = getOnlyNodeOfType(graph, 'CameraInit')

if images:
    views, intrinsics = cameraInit.nodeDesc.buildIntrinsics(cameraInit, images)
    cameraInit.viewpoints.value = views
    cameraInit.intrinsics.value = intrinsics
```

Reply:
Yes, there is no error for now but that's the usage intention.
So you're right, we can enforce the fact that it should be exclusive.

Reply:
This is changed and I have also added another argument "--import".